internal/appsec/dyngo/instrumentation/httpsec/http.go

```diff
@@ -54,16 +50,26 @@ func WrapHandler(handler http.Handler, span ddtrace.Span) http.Handler {
 	span.SetTag("_dd.runtime_family", "go")
 
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		args := MakeHandlerOperationArgs(r)
 		op := StartOperation(
-			makeHandlerOperationArgs(r, span),
+			args,
 			nil,
 		)
 		defer func() {
 			var status int
 			if mw, ok := w.(interface{ Status() int }); ok {
 				status = mw.Status()
 			}
-			op.Finish(HandlerOperationRes{Status: status})
+			events := op.Finish(HandlerOperationRes{Status: status})
```

Comment:
Improvement: the security events are now returned by the `Finish()` method - no more weird/intrusive API between the WAF and the integrations.